assets/php/virion.php

```diff
@@ -117,11 +120,13 @@ function virion_infect(Phar $virus, Phar $host, string $prefix = "", int $mode =
                 $newRel = $ligase . cut_prefix($rel, $restriction);
             }
             $data = change_dna(file_get_contents($name), $antigen, $antibody, $mode, $viralChanges); // it's actually RNA
-            $host[$newRel] = $data;
+            $hostTmpDir->addFile($newRel, $data);
         }
     }
 
-    $host["virus-infections.json"] = json_encode($infectionLog);
+    $hostTmpDir->addFile("virus-infections.json", json_encode($infectionLog));
+    $host->buildFromDirectory($hostTmpDir->path);
+    $hostTmpDir->delete();
```

Comment:
Consider putting `delete()` in a `finally` block to prevent tempdir leak

Reply:
Can a call to `delete()` in `TempDir::__destruct()` be considered as safe

Reply:
Destructors are not guaranteed to be called. I want to ensure this to prevent a CVE that drains disk space.